pllug-presentation-system/presentation-data/htmlimport.cpp

```diff
@@ -0,0 +1,40 @@
+#include "htmlimport.h"
+#include "presentationelementfactory.h"
+#include "presentationElement.h"
+
+#include <QDomDocument>
+#include <QDebug>
+#include <QByteArray>
+
+HtmlImport::HtmlImport(std::shared_ptr<PresentationElementFactory> presentationElementFactory)
+    : mPresentationElementFactory(presentationElementFactory)
+{
+}
+
+QList<PresentationElement *> HtmlImport::import(const QByteArray &inputHtml) const
+{
+    QList<PresentationElement *> rElements;
+
+    QDomDocument htmlDocument;
+    htmlDocument.setContent(inputHtml);
+
+    QDomNode domNode = htmlDocument.documentElement().firstChild();
+    while (!(domNode.isNull()))
+    {
+        QDomElement domElement = domNode.toElement();
+        if (!domElement.isNull())
+        {
+            QString htmlElement( "<" + domElement.tagName() + ">" + domElement.text() + "</" + domElement.tagName() + ">");
```

Comment:
We need to find some other method to extract full element with tag, because in this case when tag will contain some attributes they will be missing in constructed html string.